### ISSUE TITLE
Optional One to Many relationship support

### ISSUE THREAD

**Issue by omiz** [ftchirou/PredicateKit#5]
Is it possible to have a predicate for an optional one to Many relationship?

An example of what I'm trying to achieve is this
I want to get the objects without a relation 

```swift
class NoteGroup: NSManagedObject {
    var notes: [Note]?
}

class Note: NSManagedObject {
    var group: NoteGroup?
}

let predicate = (\Note.group) == nil
```

unfortunately that currently doesn't work since `NoteGroup` is not a primitive type so it can't be passed as a fetch where condition

**Comment by ftchirou** [comment]
Hey! Thanks for opening this issue. I'll look into adding support for something like this. Will update this issue with my findings.

**Comment by ftchirou** [comment]
Fixed in [`1.4.0`](https://github.com/ftchirou/PredicateKit/releases/tag/1.4.0). 

You should be able to write the following once you update to `1.4.0` @omiz.

```swift
let predicate: Predicate<Note> = \Note.group == nil
```

If you store your predicate in a variable, you may need to explicitly provide the type (as in the example above) to help the compiler disambiguate between `\Note.group == nil // => Predicate<Note>` and `\Note.group == nil // => Bool`.
